appengine-flexible-archetype/src/main/resources/archetype-resources/src/main/appengine/app.yaml

```diff
@@ -1,20 +1,23 @@
-#  Copyright 2016 Google Inc. All Rights Reserved.
-#
-#  Licensed under the Apache License, Version 2.0 (the "License");
-#  you may not use this file except in compliance with the License.
-#  You may obtain a copy of the License at
-#
-#       http://www.apache.org/licenses/LICENSE-2.0
-#
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS,
-#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#  See the License for the specific language governing permissions and
-#  limitations under the License.
-# [START appyaml]
-runtime: java
-vm: true
+##  Copyright 2016 Google Inc.
+##
+##  Licensed under the Apache License, Version 2.0 (the "License");
+##  you may not use this file except in compliance with the License.
+##  You may obtain a copy of the License at
+##
+##       http://www.apache.org/licenses/LICENSE-2.0
+##
+##  Unless required by applicable law or agreed to in writing, software
+##  distributed under the License is distributed on an "AS IS" BASIS,
+##  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+##  See the License for the specific language governing permissions and
+##  limitations under the License.
+## [START appyaml]
+runtime:    java
+env:        flex
 threadsafe: true
+#if ( $service != "default" )
+service:    $service
```

Comment:
Could you add a comment to clarify what this is for inside the appyaml? Also note that this is not set when default because it's understood according to -> https://cloud.google.com/appengine/docs/standard/python/config/appref#service